core/src/main/java/fj/Monoid.java

```diff
@@ -402,6 +402,16 @@ public A join(final Iterable<A> as, final A a) {
   }
 
   /**
+   * A monoid for the maximum of two integers.
+   */
+  public static final Monoid<Integer> intMaxMonoid = monoid(Semigroup.intMaximumSemigroup, Integer.MIN_VALUE);
```

Comment:
Do you have a use case for this? It is a valid instance but does not feel very mathematical as it rely on technicalities of integer representation limitations.
If I'm  looking for a max value, I rarely want to default to ```Integer.MIN_VALUE```: I prefer to be explicit about the default and use the semigroup.